logstash-core/src/main/java/org/logstash/StringInterpolation.java

```diff
@@ -1,92 +1,80 @@
+            @Override
+            public StringBuilder get() {
+                StringBuilder b = super.get();
+                b.setLength(0); // clear/reset the buffer
+                return b;
+            }
 
+        };
+    
     private StringInterpolation() {
-        // TODO:
-        // This may need some tweaking for the concurrency level to get better memory usage.
-        // The current implementation doesn't allow the keys to expire, I think under normal usage
-        // the keys will converge to a fixed number.
-        //
-        // If this code make logstash goes OOM, we have the following options:
-        //  - If the key doesn't contains a `%` do not cache it, this will reduce the key size at a performance cost.
-        //  - Use some kind LRU cache
-        //  - Create a new data structure that use weakref or use Google Guava for the cache https://code.google.com/p/guava-libraries/
+        // Utility Class
     }
 
-    public static void clearCache() {
-        cache.clear();
-    }
-
-    public static int cacheSize() {
-        return cache.size();
-    }
-
-    public static String evaluate(Event event, String template) throws IOException {
-        TemplateNode compiledTemplate = cache.get(template);
-
-        if (compiledTemplate == null) {
-            compiledTemplate = compile(template);
-            cache.put(template, compiledTemplate);
+    public static String evaluate(final Event event, final String template) throws IOException {
+        int open = template.indexOf("%{");
+        int close = template.indexOf('}', open);
+        if (open == -1 || close == -1) {
+            return template;
         }
-
-        return compiledTemplate.evaluate(event);
-    }
-
-    public static TemplateNode compile(String template) {
-        Template compiledTemplate = new Template();
-
-        if (template.indexOf('%') == -1) {
-            // Move the nodes to a custom instance
-            // so we can remove the iterator and do one `.evaluate`
-            compiledTemplate.add(new StaticNode(template));
-        } else {
-            Matcher matcher = TEMPLATE_TAG.matcher(template);
-            String tag;
-            int pos = 0;
-
-            while (matcher.find()) {
-                if (matcher.start() > 0) {
-                    compiledTemplate.add(new StaticNode(template.substring(pos, matcher.start())));
-                }
-
-                tag = matcher.group(1);
-                compiledTemplate.add(identifyTag(tag));
-                pos = matcher.end();
+        final StringBuilder builder = STRING_BUILDER.get();
+        int pos = 0;
+        while (open > -1 && close > -1) {
+            if (open > 0) {
+                builder.append(template, pos, open);
             }
-
-            if(pos <= template.length() - 1) {
-                compiledTemplate.add(new StaticNode(template.substring(pos)));
+            if (template.regionMatches(open + 2, "+%s", 0, 3)) {
```

Comment:
Hardcoding `3` for length gives you interesting results that are differnt from Logstash 5.5. I think you need to check the whole length (`close - open - 1` ?)



| outcome | input | v5.5.0 | this pr | 
|-|-|-|-|
| same | +%s | 1500508897 | 1500509113 |
| different | +%ss | %43 | 1500509162 |
| different | +%s d y | %30 20 2017 | 1500509234 |

In all cases above, since the first 3 chars are "%+s" it assumes unix epoch time and ignores the rest.